source/agora/script/Stack.d

```diff
@@ -0,0 +1,233 @@
+/*******************************************************************************
+
+    Contains a stack implementation for use with the script execution engine.
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.script.Stack;
+
+import agora.common.Serializer;
+
+import std.container : SList;
+import std.range;
+
+version (unittest)
+{
+    import ocean.core.Test;
+    import std.stdio;
+}
+
+/// Maximum total stack size
+public enum MAX_STACK_TOTAL_SIZE = 16_384;
```

Comment:
`MAX_STACK_*` => `STACK_MAX_*` for better visual categorization. Also, we can just move it to consensus-style constructor parameters.